### ISSUE TITLE
Embedding fails with certain indices.

### ISSUE THREAD

**Issue by llongTT** [tenstorrent/tt-metal#18676]
As described in title:

@pytest.mark.parametrize(
    "embedding_specs, dtype, indices_layout, weight_layout, output_layout",
    [
        (
            {"weight_shape": [81, 768], "indices_shape": [1, 24], "padding_idx": 1},
            ttnn.bfloat16,
            ttnn.TILE_LAYOUT,
            ttnn.ROW_MAJOR_LAYOUT,
            ttnn.ROW_MAJOR_LAYOUT,
        ),
    ],
)
def test_embedding_single(
    embedding_specs,
    dtype,
    indices_layout,
    weight_layout,
    output_layout,
    *,
    device,
):
    device.enable_async(False)

    # Extract the weight and indices shape from embedding_specs
    weight_shape = embedding_specs["weight_shape"]
    indices_shape = embedding_specs["indices_shape"]
    padding_idx = embedding_specs.get("padding_idx", None)  # Optional padding index

    # Create random weight and indices tensors in PyTorch
    weight = torch_random(weight_shape, -0.1, 0.1, dtype=torch.bfloat16)
    indices = torch.randint(0, weight_shape[0], indices_shape, dtype=torch.int32)
    indices =  torch.tensor([[74, 16, 25,  0, 79, 33, 44, 65, 71, 48, 42, 73, 12, 31, 34, 70, 21, 50,
         66, 71, 19,  1, 22,  1]], dtype=torch.int32)
    print("indices = ", indices)

    # Create a PyTorch embedding layer and apply it
    torch_embedding = torch.nn.Embedding.from_pretrained(weight, padding_idx=padding_idx)
    torch_output_tensor = torch_embedding(indices)

    # Convert the weight and indices to ttnn tensor format
    ttnn_weight = ttnn.from_torch(weight, device=device, layout=weight_layout, dtype=dtype)
    ttnn_indices = ttnn.from_torch(indices, device=device, layout=indices_layout, dtype=ttnn.uint32)

    # Measure performance of the embedding operation in ttnn
    start_time = start_measuring_time()
    print(
        "weight_shape={}, indices_shape={}, padding_idx ={}, indices_layout={}, weight_layout={}, output_layout={}.llong".format(
            weight_shape, indices_shape, padding_idx, indices_layout, weight_layout, output_layout
        )
    )
    try:
        # Apply embedding in ttnn
        ttnn_output_tensor = ttnn.embedding(
            ttnn_indices,
            ttnn_weight,
            padding_idx=padding_idx,
            layout=output_layout,
            embeddings_type=ttnn.EmbeddingsType.GENERIC,  # Default embeddings type
            dtype=dtype,
            output_tensor=None,  # No preallocated output tensor
            memory_config=None,  # Default memory config
            queue_id=0,  # Default queue id
        )

        e2e_perf = stop_measuring_time(start_time)

        # Convert the ttnn tensor back to PyTorch for comparison
        ttnn_output_tensor = ttnn.to_torch(ttnn_output_tensor)
    except Exception as e:
        print("An error occurred:", e)
        # print("weight_shape={}, indices_shape={}, padding_idx ={}, indices_layout={}, weight_layout={}, output_layout={}.llong".format(weight_shape, indices_shape, padding_idx, indices_layout,weight_layout, output_layout))
        traceback.print_exc()  # Print the full stack trace
        return [False, None]
    result, msg = check_with_pcc(torch_output_tensor, ttnn_output_tensor, 0.999)
    print("run result: {}, with message= {} ".format(result, msg))
    assert_with_pcc(torch_output_tensor, ttnn_output_tensor, 0.999)

**Comment by ntarafdar** [comment]
closing as stale
